src/sentry/options/manager.py

```diff
@@ -369,3 +450,47 @@ def get_last_update_channel(self, key: str) -> Optional[UpdateChannel]:
         # be applied evaluating all the possible drift cases.
         opt = self.lookup_key(key)
         return self.store.get_last_update_channel(opt)
+
+    def can_update(self, key: str, value, channel: UpdateChannel) -> Optional[NotWritableReason]:
+        """
+        Return the reason the provided channel cannot update the option
+        to the provided value or None if there is no reason and the update
+        is allowed.
+        """
+
+        required_flag = WRITE_REQUIRED_FLAGS.get(channel)
+        opt = self.lookup_key(key)
+        if opt.flags & (FLAG_NOSTORE | FLAG_IMMUTABLE):
+            return NotWritableReason.READONLY
+        if opt.flags & FLAG_PRIORITIZE_DISK and settings.SENTRY_OPTIONS.get(key):
+            return NotWritableReason.OPTION_ON_DISK
+
+        if required_flag and not (opt.flags & required_flag):
+            return NotWritableReason.CHANNEL_NOT_ALLOWED
+
+        if not self.isset(key):
+            # If the option is not readonly and it is not stored in the
+            # option store it means we are relying on default. So we can
+            # update.
+            return None
+
+        stored_value = self.get(key)
+        if stored_value == value:
+            # In theory options could have any type so this equality may
+            # not be correct.
+            # In practice, this code is added to support the move towards
+            # configMap backed options, which will be restricted to types
+            # that allow for this equality: basic types, sets, list, maps.
+            # So even if this equality fails, in the worst case scenario
+            # we would not allow the update if there is a mismatch between
+            # the channels.
```

Comment:
This condition can have false negatives?  If the equality check fails then an option that would otherwise be update-able might return as `NotWritableReason.DRIFTED`?  If you can't set `key=existing_value` to change the last update channel whats the workaround?  Re-deploy?